compiler-rt/lib/rtsan/rtsan_assertions.h

```diff
@@ -14,15 +14,24 @@
 
 #include "rtsan/rtsan.h"
 #include "rtsan/rtsan_context.h"
+#include "sanitizer_common/sanitizer_stacktrace.h"
 
 namespace __rtsan {
 
 template <typename OnViolationAction>
-void ExpectNotRealtime(Context &context, OnViolationAction &&OnViolation) {
+void ExpectNotRealtime(Context &context, OnViolationAction &&OnViolation,
+                       __sanitizer::uptr caller_pc,
+                       __sanitizer::uptr caller_bp) {
   CHECK(__rtsan_is_initialized());
   if (context.InRealtimeContext() && !context.IsBypassed()) {
     context.BypassPush();
-    OnViolation();
+
+    __sanitizer::BufferedStackTrace stack;
+    stack.Unwind(caller_pc, caller_bp, nullptr,
+                 /*fast*/ true);
```

Comment:
Getting the stack here is essential for deduplication. See the next change in the series here:

https://github.com/cjappl/llvm-project/pull/9/files#diff-26b1e9b52cc6bd23ad235825b8438ded1867285654d29b60536c5e87876adb3cR39-R46